src/Simulation/Simulators.Tests/TestProjects/HoneywellExe/ClassicallyControlledSupportTests.qs

```diff
@@ -0,0 +1,732 @@
+        }
+    }
+
+    operation BothInvalid() : Unit {
+        let r = Zero;
+        if (r == Zero) {
+            SubOp1();
+            SubOp2();
+            return ();
+        } else {
+            SubOp2();
+            SubOp3();
+            return ();
+        }
+    }
+    
+    operation ApplyIfZero_Test() : Unit {
+        let r = Zero;
+        if (r == Zero) {
+            SubOp1();
+        }
+    }
+
+    operation ApplyIfOne_Test() : Unit {
+        let r = Zero;
+        if (r == One) {
+            SubOp1();
+        }
+    }
+
+    operation ApplyIfZeroElseOne() : Unit {
+        let r = Zero;
+        if (r == Zero) {
+            SubOp1();
+        } else {
+            SubOp2();
+        }
+    }
+
+    operation ApplyIfOneElseZero() : Unit {
+        let r = Zero;
+        if (r == One) {
+            SubOp1();
+        } else {
+            SubOp2();
+        }
+    }
+
+    operation IfElif() : Unit {
+        let r = Zero;
+
+        if (r == Zero) {
+            SubOp1();
+        } elif (r == One) {
+            SubOp2();
+        } else {
+            SubOp3();
+        }
+    }
+
+    operation AndCondition() : Unit {
+        let r = Zero;
+        if (r == Zero and r == One) {
+            SubOp1();
+        } else {
+            SubOp2();
+        }
+    }
+
+    operation OrCondition() : Unit {
+        let r = Zero;
+        if (r == Zero or r == One) {
+            SubOp1();
+        } else {
+            SubOp2();
+        }
+    }
+
+    operation ApplyConditionally() : Unit {
+        let r1 = Zero;
+        let r2 = Zero;
+        if (r1 == r2) {
+            SubOp1();
+        }
+        else {
+            SubOp2();
+        }
+    }
+
+    operation ApplyConditionallyWithNoOp() : Unit {
+        let r1 = Zero;
+        let r2 = Zero;
+        if (r1 == r2) {
+            SubOp1();
+        }
+    }
+
+    operation InequalityWithApplyConditionally() : Unit {
+        let r1 = Zero;
+        let r2 = Zero;
+        if (r1 != r2) {
+            SubOp1();
+        }
+        else {
+            SubOp2();
+        }
+    }
+
+    operation InequalityWithApplyIfOneElseZero() : Unit {
+        let r = Zero;
+        if (r != Zero) {
+            SubOp1();
+        }
+        else {
+            SubOp2();
+        }
+    }
+
+    operation InequalityWithApplyIfZeroElseOne() : Unit {
+        let r = Zero;
+        if (r != One) {
+            SubOp1();
+        }
+        else {
+            SubOp2();
+        }
+    }
+
+    operation InequalityWithApplyIfOne() : Unit {
+        let r = Zero;
+        if (r != Zero) {
+            SubOp1();
+        }
+    }
+
+    operation InequalityWithApplyIfZero() : Unit {
+        let r = Zero;
+        if (r != One) {
+            SubOp1();
+        }
+    }
+
+    operation LiteralOnTheLeft() : Unit {
+        let r = Zero;
+        if (Zero == r) {
+            SubOp1();
+        }
+    }
+
+    //operation GenericsSupport<'A, 'B, 'C>() : Unit {
```

Comment:
What is the reason this is commented out? Not that that would be an issue - just curious.

Reply:
When this is not commented out, even without the corresponding test, I get the error:
CS0305	Using the generic type '_0937a8eea43e4297a95fbffc1a4e8b2f_GenericsSupport<__A__, __B__, __C__>' requires 3 type arguments.

Reply:
That is something that should be fixed then, though - depending on what the source of the issue is as part of this PR (if it is related the the targeting) or in a separate PR (if it is unrelated and also occurs for a project that is not targeted). Could you please investigate which one it is?

Reply:
This issue is not related to targeting, but seems to be a bug in the C# generation for generic callables. I will file an issue.

Reply:
Here is the issue for this: https://ms-quantum.visualstudio.com/Quantum%20Program/_workitems/edit/17245
I'm going to close out this conversation as this is not related to the purpose of this PR.

Reply:
I've also added ToDos in the code for this.

Reply:
Since non-employees can't see ADO work items, could you create an issue on the qsharp-runtime GitHub repo?

Reply:
I've added the following issue for this: #299

Reply:
@ScottCarda-MS The issue is related to targeting as the same code compiles fine if you don't target the project.